### DISCUSSION TITLE
Could you provide a method to get random value k in DSA/ECDSA?

### DISCUSSION THREAD

**Discussion by TSAlicorn** [openssl/openssl#24570] [Q&A] [answered]
I'm a verification engineer, and I am using openssl lib as golden data to test some security algorithm implemented in hardware like DSA/ECDSA.
I send params generated by openssl to design and compare the output with golden data.
But the input of design need the random value _k_ inside `dsa_sign_setup` method, and it seems that impossible to get it directly by calling a specific method.
So I have to modify the code and save the _k_ value to a file, then I can read the file to get the _k_ value. 
It would be great if you can provide corresponding method to help many hardware engineers like me, thank you!

**Comment by nhorman** [comment, marked as answer]
can't this be done in reverse order?  i.e. the openssl gendsaparam applet contains a --writerand option that allows you, when generating dsa parameters to write out the corresponding private random data to a separate file, which I believe represents the k value those params use.

alternatively, I believe you can use the hexseed parameter to specify the k value to use for a given input when generating parameters, or the OSSL_PKEY_PARAM_FFC_SEED parameter, if you want to do so programatically.

**Reply by TSAlicorn** [reply]
Thank you for your reply. I indeed want to obtain the _k_ value through C code. The `OSSL_PKEY_PARAM_FFC_SEED` parameter seems to be just a random seed, and I have not found a way to convert it into the _k_ value. 
And from the description, it seems that the parameter is only related to the _p_, _q_, _g_ values.
